src/app/geral-module/main/main.component.ts

```diff
@@ -29,6 +29,10 @@ export class MainComponent implements OnInit {
       {
         label: 'Progresso da turma',
         command: () => { this.router.navigate(["main", { outlets: { principal: ['progressoTurma'] } }]) }
+      },
+      {
+        label: 'Logout',
+        command: () => {this.router.navigate([""])}
```

Comment:
Deve ser feito um método no main.component para realizar o logout e não redirecionar para a tela de login. Pois da forma como está feito, ao entrar na tela de login o cara é deslogado, sendo que se você está conectado e vai para o login não deveria deslogar, e sim redirecionar para o main (como acontece no Facebook, por exemplo)

Reply:
Fiz a função no main.component:

private logout() {
    if(Usuario.logout()){
      return this.router.navigate([""])
    }
}

Tentei utilizar o this.router.url para fazer o mecanismo de quando o usuário estiver logado e for acessar a login: http://localhost:4200 ele retornar para o http://localhost:4200/main. Porém quando dou print na url só printa: /. E o elemento auth.guard só é chamado uma vez.

Tentei ajustar a função para que o usuário ao colocar a url: "" (login), fosse renderizado para o "main" da seguinte forma:

verificarAcesso() {
    if(Usuario.isUsuarioLogado() && this.router.url == "") {
      this.router.navigate(["/main"]);
      return false;
    }else if (Usuario.isUsuarioLogado()) {
      return true;
    }
      this.router.navigate([""]);
      return false;
  }

Porém dessa foram entre em loop.